### DISCUSSION TITLE
Was minio/minio:RELEASE.2021-05-27T22-06-31Z removed from docker hub?

### DISCUSSION THREAD

**Discussion by mboutet** [minio/minio#12479] [Q&A] [answered]
Hi, I have `minio/minio:RELEASE.2021-05-27T22-06-31Z` deployed in a Kubernetes cluster, but suddenly the image can no longer be found on docker hub.

Any idea?

**Comment by harshavardhana** [comment, marked as answer]
It was removed on purpose due to a bug discovered mentioned in the release notes  https://github.com/minio/minio/releases/tag/RELEASE.2021-06-07T21-40-51Z it is to avoid surprises for users upgrading from older releases.

Those releases have been quarantined, to avoid any further issues down the line as a precautionary measure.

You can move to the latest release instead.

**Reply by mboutet** [reply]
Ok, I understand the reason, but it might be better to treat an image pushed to docker hub as immutable and permanent even though it contains bugs and/or security issues.

Removing or altering an image from a registry might cause outages for users which is really bad and worse than having an images with bugs. At least, it gives users some time to switch to the new image and validate that everything works as intended.

I would seriously consider not doing that in the future if I were you because the blast radius can potentially be huge.

And, yes, it would be best that such users like myself use a private registry that mirrors the images, but not everyone has the time and luxury to set it up. Hence, these users rely on the maintainer of a docker hub namespace to not remove or alter images.

Thank you.

**Reply by harshavardhana** [reply]
There is no reason to have that kind of hard-line here - we never really do it, but we have to do it unfortunately in this scenario. We have 1000's of users for many years and we cannot control what they upgrade to and the situation mainly affects those users.

It is in everyone's interest to remove the images that are affected by it because there is no real way to control what everyone is doing in their local infrastructure. 

if you are relying on the image you should have a local copy of it already unless you are doing "ImagePull: Always" which is generally considered a bad practice. If you see an image is missing there is a good reason we have done it so to avoid further damage. 

> Removing or altering an image from a registry might cause outages for users which is really bad and worse than having an images with bugs. At least, it gives users some time to switch to the new image and validate that everything works as intended.

In this scenario we have to disagree this is not the usual run of the mill bug that you can validate locally, object corruption is an intricate problem and cannot be recovered from so it is important that you took notice and move to the right release instead of never knowing what you were running in the first place.

**Reply by mboutet** [reply]
Thanks for the detail. I guess I'm less affected by the bug you mentioned because I only use minio in gateway mode.
